addons/de/xrechnung/invoices.go

```diff
@@ -0,0 +1,198 @@
+package xrechnung
+
+import (
+	"github.com/invopop/gobl/bill"
+	"github.com/invopop/gobl/cbc"
+	"github.com/invopop/gobl/org"
+	"github.com/invopop/validation"
+)
+
+const (
+	invoiceTypeSelfBilled               cbc.Key = "self-billed"
+	invoiceTypePartial                  cbc.Key = "partial"
+	invoiceTypePartialConstruction      cbc.Key = "partial-construction"
+	invoiceTypePartialFinalConstruction cbc.Key = "partial-final-construction"
+	invoiceTypeFinalConstruction        cbc.Key = "final-construction"
+)
+
+var validTypes = []cbc.Key{
+	bill.InvoiceTypeStandard,
+	bill.InvoiceTypeCreditNote,
+	bill.InvoiceTypeCorrective,
+	invoiceTypeSelfBilled,
+	invoiceTypePartial,
+	invoiceTypePartialConstruction,
+	invoiceTypePartialFinalConstruction,
+	invoiceTypeFinalConstruction,
+}
```

Comment:
Rather than modify the invoice types (which is strictly prohibited), checkout the usage of "tags" and "scenarios" in other addons.